core/state_transition.go

```diff
@@ -206,6 +218,20 @@ func (st *StateTransition) preCheck() error {
 				st.msg.From().Hex(), msgNonce, stNonce)
 		}
 	}
+	// Make sure baseFee is only defined for blocks after the Aleut fork, and that
+	// the transaction feeCap is greater than the baseFee.
+	isAleut := st.evm.ChainConfig().IsAleut(st.evm.Context.BlockNumber)
+	switch {
+	case isAleut && st.evm.Context.BaseFee != nil:
+		if st.feeCap.Cmp(st.evm.Context.BaseFee) < 0 {
+			return fmt.Errorf("%w: address %v, feeCap: %d baseFee: %d", ErrFeeCapTooLow,
+				st.msg.From().Hex(), st.feeCap.Uint64(), st.evm.Context.BaseFee.Uint64())
+		}
+	case isAleut && st.evm.Context.BaseFee == nil:
+		return fmt.Errorf("baseFee missing")
+	case !isAleut && st.evm.Context.BaseFee != nil:
```

Comment:
Looking back at this, do people think it makes sense to check that if EIP-1559 is not enable then the `basefee` is `nil`? Can't seem to find it now, but I recall someone mentioning that they weren't fond of the `BASEFEE` op potentially dereferencing `nil` if *somehow* it made it past the fork checks and included the op in the jump table.

The reason I'm coming back to this is it seems `t8n` circumvents some of the code in `state_processor` so it can create blocks with no `nil` `basefee` even if `Aleut` is configured.

Reply:
All of the errors here are named errors, like `ErrNonceTooHigh`, so the outer calller can do `errors.Is(err, ErrNonceTooHigh)`. I think we should define these errors too

Reply:
See `core/state_processor_test.go`: `TestStateProcessorErrors`. We should extend those tests for these cases.